crates/phactory/src/system/gk.rs

```diff
@@ -56,6 +57,18 @@ fn next_random_number(
     hashing::blake2_256(buf.as_ref())
 }
 
+fn get_contract_key(
+    master_key: &sr25519::Pair,
+    contract_info: &phala_types::contract::ContractInfo<chain::Hash, chain::AccountId>,
+) -> sr25519::Pair {
+    master_key
+        .derive_sr25519_pair(&[
+            b"contract_key",
+            Encode::encode(contract_info).as_bytes_ref(),
```

Comment:
Not a good idea to use the full contract info to derive the key. A reasonable ways is to generate a random number (e.g. use the block number + event index as the derive path). If we want to refer to the contract instance, I suggest to use the global contract id (after Kevin's PR to the upstream is merged and in use)

Reply:
OK to leave a TODO here

Reply:
done